doc/developer_documentation.md

```diff
@@ -650,3 +651,16 @@ Run the following command from the repo root to auto-format all C++ code:
 ```bash
 find src | grep "\.\(cc\|h\)$" | xargs clang-format -i
 ```
+
+# <a name="newfile"></a>Adding new C++ files
+
+For the cmake build system, we maintain C++ file lists in the folder `file_lists`.
+When you add a new C++ file, they can be updated as follows:
+
+```bash
+# from repo root
+find src | grep \.cc | grep -Ev "test|pybind|perf|^src/main\.cc" > file_lists/source_files_no_main
```

Comment:
Add a script to the `dev/` directory that does this (in bash assuming linux is fine).

Add a step to the `ci.yml` generated-files-up-to-date check that verifies the files are up to date. (Search for "diff" in the ci.yml file).

Reply:
done!